src/providers/cortex/trulens/providers/cortex/provider.py

```diff
@@ -9,38 +8,100 @@
 from trulens.providers.cortex.endpoint import CortexEndpoint
 
 
-class Cortex(LLMProvider):
-    # require `pip install snowflake-snowpark-python` and a active Snowflake account with proper privileges
+class Cortex(
+    LLMProvider
+):  # require `pip install snowflake-snowpark-python` and a active Snowflake account with proper privileges
     # https://docs.snowflake.com/en/user-guide/snowflake-cortex/llm-functions#availability
 
     DEFAULT_MODEL_ENGINE: ClassVar[str] = "snowflake-arctic"
 
+    connection_parameters: Dict
     model_engine: str
+
     """Snowflake's Cortex COMPLETE endpoint. Defaults to `snowflake-arctic`.
        Reference: https://docs.snowflake.com/en/sql-reference/functions/complete-snowflake-cortex
+
+    Args:
+
+        connection_parameters (Dict): Snowflake connection parameters.
+        model_engine (str, optional): Model engine to use. Defaults to `snowflake-arctic`.
+
+        Connecting with user/password:
+
+        !!! example
+
+            ```python
+            connection_parameters = {
+
+                "account": <account>,
+                "user": <user>,
+                "password": <password>,
+                "role": <role>,
+                "database": <database>,
+                "schema": <schema>,
+                "warehouse": <warehouse>
+            }
+
+            provider = Cortex(connection_parameters)
+            ```
+
+        Connecting with private key:
+
+        !!! example
+
+            ```python
+            connection_parameters = {
+
+                "account": <account>,
+                "user": <user>,
+                "private_key": <private_key>,
+                "role": <role>,
+                "database": <database>,
+                "schema": <schema>,
+                "warehouse": <warehouse>
+            }
+
+        Connecting with a private key file:
+
+        !!! example
+
+            ```python
+            connection_parameters = {
+
+                    "account": <account>,
+                    "user": <user>,
+                    "private_key_file": <private_key_file>,
+                    "private_key_file_pwd": <private_key_file_pwd>,
+                    "role": <role>,
+                    "database": <database>,
+                    "schema": <schema>,
+                    "warehouse": <warehouse>
+                }
+            provider = Cortex(connection_parameters)
+            ```
     """
 
     endpoint: CortexEndpoint
     snowflake_conn: SnowflakeConnection
 
     def __init__(
-        self, model_engine: Optional[str] = None, *args, **kwargs: Dict
+        self,
+        connection_parameters: Dict,
```

Comment:
this (adding `connection_parameters` to the constructor` currently breaks the pydantic validation. I'll put up a PR to fix. cc @joshreini1 @sfc-gh-chu